core/server/services/auth/api-key/content.js

```diff
@@ -0,0 +1,43 @@
+const models = require('../../../models');
+const common = require('../../../lib/common');
+
+const authenticateContentApiKey = function authenticateContentApiKey(req, res, next) {
+    if (req.headers && req.headers.authorization) {
+        return next(new common.errors.BadRequestError({
+            message: 'Content API does not support header authentication',
+            code: 'INVALID_AUTH_TYPE'
+        }));
+    }
+
+    // allow fallthrough to other auth methods or final ensureAuthenticated check
+    if (!req.query || !req.query.content_key) {
+        return next();
+    }
+
+    let key = req.query.content_key;
+
+    models.ApiKey.findOne({secret: key}).then((apiKey) => {
+        if (!apiKey) {
+            return next(new common.errors.UnauthorizedError({
+                message: 'Unknown Content API Key',
+                code: 'UNKNOWN_CONTENT_API_KEY'
+            }));
+        }
+
+        // TODO: extra messaging in case Admin secret was exposed?
+        if (apiKey.get('type') !== 'content') {
+            return next(new common.errors.UnauthorizedError({
+                message: 'Incorrect API Key type',
+                code: 'INCORRECT_API_KEY_TYPE'
+            }));
+        }
+
+        // authenticated OK, store the api key on the request for later checks and logging
+        req.api_key = apiKey;
+        next();
+    });
```

Comment:
Do we need to pass `next` here too? or in `catch`?

Reply:
Sorry, I'm not sure I'm following 😬

Reply:
Sorry - doesn't make sense on that line! I mean, we only call next if that promise resolves, should we be passing `next` as either the second param to `then` or calling `catch` on the promise returned from `findOne`?

Reply:
Yeah in fact an actual internal server error is much better :sweat_smile: